src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControl.java

```diff
@@ -106,6 +123,17 @@
     }
 
     /**
+     * Construct a child node. The concatenation of regular expressions needs special care:
+     * see {@link #ensureSelfContainedRegex(String, boolean)} for more details.
+     * @param parent the parent node.
+     * @param subPkg the sub package name.
+     * @param regex flags interpretation of subPkg as regex pattern.
+     */
+    ImportControl(final ImportControl parent, final String subPkg, final boolean regex) {
```

Comment:
@romani There are valid uses of `final` on parameters or for-loop variables. If it is used in an anonymous class, it must be declared final or it can be referenced directly inside the class.
That said, we should have a check that produces a violation if it isn't used in an anonymous class.